iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -164,11 +164,11 @@ PoshRuntimeImpl::requestPublisherFromRoudi(const IpcMessage& sendBuffer) noexcep
         if (stringToIpcMessageType(IpcMessage.c_str()) == IpcMessageType::CREATE_PUBLISHER_ACK)
 
         {
-            rp::BaseRelativePointer::id_underlying_t segmentId{0U};
+            rp::segment_id_underlying_t segmentId{0U};
```

Comment:
This name space pollution can become problematic. All these types which are implementation details of the relative pointer but are exposed are in the `rp` namespace, maybe later `memory`. 

I have no good answer how to solve this though (specific detail namespace for each class/unit?) . It is no (feasible) option though to let them depend on a class with template parameters, as there would be different definitions for each template parameter.

Reply:
I mean that the scope of something we define should be only as large as needed (nothing about build-time). This is mainly to discourage usage in other unrelated classes and is basically about cohesion: keep things that belong together together and separate it from things they are not related to.

Ideally it would be a type that is tied to the only class that uses it, if there is such a class. If its only used in class A, it should be defined there. If the client needs to know it, it should be public but otherwise private etc. This is not possible here. It is a minor issue while there are few classes but becomes a headache when it grows.